app/controllers/devise_sessions_controller.rb

```diff
@@ -88,13 +118,37 @@ def verify_signed_out_user; end
 
   def check_login_state
     @login_state_id = session[:login_state_id]
-    redirect_to new_user_session_path unless login_state
+    redirect_to user_session_path unless login_state
   end
 
   def check_password_ok
     redirect_to user_session_path unless login_state.password_ok
   end
 
+  def after_login_path(payload, user)
+    payload&.dig(:post_login_oauth).presence || after_sign_in_path_for(user)
+  end
+
+  def create_jwt(payload)
+    jwt = Jwt.create!(
+      jwt_payload: payload,
+    )
+
+    session[:jwt_id] = jwt.id
+  end
+
+  def create_login_state(payload, email, password_ok = false)
```

Comment:
Similar comment as above: is it worth inlining this into the only method which uses it?